### ISSUE TITLE
Auto release scripts are broken

### ISSUE THREAD

**Issue by emlys** [natcap/invest#658]
I have run into a few issues with the auto release scripts (`scripts/release-1-initiate.sh`, `scripts/release-2-commit.sh`, `scripts/release-3-publish.sh`). Some typos and some may be differences with running on macOS. It will be worth going back through once the release is done and making sure we have working scripts for windows and mac.

**Comment by emlys** [comment]
Wrapping it in a docker container could be a solution.

**Comment by dcdenu4** [comment]
I like the docker container idea and it reminds me that I'd like to get back to creating an InVEST docker container that we maintain.

**Comment by dcdenu4** [comment]
Having just done release 3.9.2 I have some feedback on these scripts.
- `release-1-initiate.sh` worked as expected on Windows
- `release-2-commit.sh` has a missing quotation bug
- `release-3-publish.sh` did not work as expected and might still be set up for automation through GHA
    - From James:
    - >Ah yes, GITHUB_TOKEN would normally be defined for us if we were in a github actions run (which we were, originally), and yes, there’ll be a personal access token you’ll need to create with the correct permisisons

- [ ] Fix `release-2-commit.sh` has a missing quotation bug
- [ ] Rework / document `release-3-publish.sh` to function as expected

**Comment by emlys** [comment]
I'm going to prioritize this now since the 3.12.1 release was missing binaries on github, after I thought I'd run the release scripts successfully.

Both for developing this script and for testing out future releases, it would be nice to have a "testpypi for invest" - a mirrored github repo where we can create test releases. This might need a proposal doc depending on how complex it gets.
